.github/workflows/reusable-unit-tests.yml

```diff
@@ -59,8 +59,11 @@ jobs:
         with:
           fetch-depth: 0
 
-      - name: setup jdk${{ inputs.jdk }}
-        run: echo "JAVA_HOME=$JAVA_HOME_${{ inputs.jdk }}_X64" >> $GITHUB_ENV
+      - uses: actions/setup-java@v3
+        with:
+          distribution: 'zulu'
+          java-version: ${{ inputs.jdk }}
+          cache: 'maven'
```

Comment:
We're managing our own caching in the `maven-restore` step, so I think we should remove `cache: 'maven'` here. While doing that could you also add a comment to our `maven-restore` action explaining why we use `github.sha` rather than a hash of the POMs like `setup-java` would? (It's because we install snapshots of our own jars to the local repo, so we can't reuse it across commits)